sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/ConstantColumnVector.java

```diff
@@ -0,0 +1,212 @@
+package org.apache.spark.sql.execution.vectorized;
+
+import org.apache.spark.sql.types.DataType;
+import org.apache.spark.sql.types.Decimal;
+import org.apache.spark.sql.vectorized.ColumnVector;
+import org.apache.spark.sql.vectorized.ColumnarArray;
+import org.apache.spark.sql.vectorized.ColumnarMap;
+import org.apache.spark.unsafe.types.UTF8String;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+
+/**
+ * This class adds the constant support to ColumnVector.
+ * It supports all the types and contains put APIs,
+ * which will put the exact same value to all rows.
+ *
+ * Capacity: The vector only stores one copy of the data, and acts as an unbounded vector
+ * (get from any row will return the same value)
+ */
+public class ConstantColumnVector extends ColumnVector {
```

Comment:
I am thinking whether we should extend `WritableColumnVector` instead, so we can easily leverage constant column vector to represent partition columns.

It seems for partition columns, we are doing copying of same value per row ([Parquet](https://github.com/apache/spark/blob/master/sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedParquetRecordReader.java#L258) and [ORC](https://github.com/apache/spark/blob/master/sql/core/src/main/java/org/apache/spark/sql/execution/datasources/orc/OrcColumnarBatchReader.java#L171)). A future improvement is to use the constant column vector we are introducing here to avoid unnecessary operations.

@cloud-fan WDYT?

Reply:
I was thinking to extend `WritableColumnVector` initially, but seems like we needs to implement some unnecessary public methods like: `putLongs(rowId, count, value)`